src/config.rs

```diff
@@ -140,20 +149,40 @@ impl Config {
             );
             self.recovery_threads = MIN_RECOVERY_THREADS;
         }
-        if !Version::is_valid(self.format_version) {
-            warn!(
-                "format-version ({}) is invalid, setting it to {}",
-                self.format_version,
-                Version::default() as u64
-            );
-            self.format_version = Version::default() as u64;
+        if self.enable_log_recycle {
+            if !self.format_version.has_log_signing() {
+                return Err(box_err!(
+                    "format_version: {:?} is invalid when 'enable_log_recycle' on, setting it to V2",
+                    self.format_version
+                ));
+            }
+            if self.purge_threshold.0 / self.target_file_size.0 >= std::u32::MAX as u64 {
+                return Err(box_err!(
+                    "File count exceed UINT32_MAX, calculated by 'purge-threshold / target-file-size'"
+                ));
+            }
         }
         #[cfg(not(feature = "swap"))]
         if self.memory_limit.is_some() {
             warn!("memory-limit will be ignored because swap feature is not enabled");
         }
         Ok(())
     }
+
+    /// Returns the capacity for recycling log files.
+    pub fn recycle_capacity(&self) -> usize {
```

Comment:
```suggestion
    pub(crate) fn recycle_capacity(&self) -> usize {
```

Reply:
Here, about the `crate`, is it neccessary for `recycle_capacity` to be accessed by outer func or callers?

Reply:
`pub(crate)` can be accessed by functions in raft-engine, but not by library user such as tikv.